OUDS/Core/Components/Sources/Internal/View+readSize.swift

```diff
@@ -0,0 +1,39 @@
+//
+// Software Name: OUDS iOS
+// SPDX-FileCopyrightText: Copyright (c) Orange SA
+// SPDX-License-Identifier: MIT
+// 
+// This software is distributed under the MIT license,
+// the text of which is available at https://opensource.org/license/MIT/
+// or see the "LICENSE" file for more details.
+// 
+// Authors: See CONTRIBUTORS.txt
+// Software description: A SwiftUI components library with code examples for Orange Unified Design System 
+//
+
+import SwiftUI
+
+// MARK: - Sizes management
+
+private struct SizePreferenceKey: @preconcurrency PreferenceKey {
+    @MainActor static var defaultValue: CGSize = .zero
+
+    static func reduce(value _: inout CGSize, nextValue _: () -> CGSize) {}
+}
+
+extension View {
+
+    /// Use to read the size of the current view (usefull to read the width of text)
+    ///
+    /// - Parameter onChange: Called when the size changes
+    @MainActor
+    func readSize(onChange: @escaping (CGSize) -> Void) -> some View {
+        background(
+            GeometryReader { geometryProxy in
+                Color.clear
+                    .preference(key: SizePreferenceKey.self, value: geometryProxy.size)
+            }
+        )
+        .onPreferenceChange(SizePreferenceKey.self, perform: onChange)
```

Comment:
J'ai un _warning_ au sujet de `onPreferenceChange` : 

*Passing non-sendable parameter 'onChange' to function expecting a @Sendable closure*

J'ai mis un patch Git en pièce jointe pour corriger (*git apply 550-572-concurrency.patch*)

[550-572-concurrency.patch](https://github.com/user-attachments/files/19547582/550-572-concurrency.patch)